lib/cisco_node_utils/cmd_ref/bridge_domain.yaml

```diff
@@ -3,11 +3,54 @@
 ---
 _exclude: [N3k, N5k, N6k, N8k, N9k, ios_xr]
 
+# For the below name, shutdown commands we are going to use show running-config
+# bridge-domain cli which displays as below
+#
+# system bridge-domain 100-110
+#
+# bridge-domain 100-110
+#
+# bridge-domain 100
+#   shutdown
+#   fabric-control
+#   name PepsiCo
+# bridge-domain 101
+# bridge-domain 102
+# ...
+# bridge-domain 108
+#   shutdown
+#   name Amazon
+# bridge-domain 109
+#   shutdown
+# bridge-domain 110
+#   name Microsoft
+
+_template:
+  get_command: "show running-config bridge-domain"
+
 all_bds:
   multiple: true
-  get_command: "show running-config bridge-domain"
   get_value: '/^bridge-domain (\S+)/'
 
+# Here for name match if condition is used because the underlying property is
+# system bridge-domain 100
+#
+# bridge-domain 100
+#
+# bridge-domain 100
+#   shutdown
+#   fabric-control
+#   name PepsiCo
+# so we cannot use get_context: '/^bridge-domain 100/' because it will search
+# for the first context and there is nothing under that. The properties are
+# present in the second context. Also name could appear in any order and hence
+# the if conditions.
+bd_name:
+  get_value: '/^bridge-domain[\s\S]*bridge-domain <bd>\s*(?:shutdown\s+fabric-control\s+name (\S+)|fabric-control\s+name (\S+)|shutdown\s+name (\S+)|name (\S+))/'
```

Comment:
This works as long as bridge-domain only has a combination of these three keywords in use. I see add'l keywords in the 7k cli so I would expect this to break if any of those keywords are used.

Our command ref infrastructure doesn't currently support multi-line matching so that's not an option yet.

For now I think the solution is to take advantage of the trailing whitespace on the second bridge-domain that nvgens, i.e. the show run for this config:

```
system bridge-domain 100-113 

bridge-domain 100

bridge-domain 100 
  shutdown 
  fabric-control
  name fred100
```

...returns this pattern:

```
...\nbridge-domain 100\n\nbridge-domain 100 \n...
```

...so we can use that to differentiate the two and write our yaml like this:

```
_template:
  get_command: "show running-config bridge-domain"
  context: '/bridge-domain <bd>\s$/'
  
name:
  get_value: '/name (\S+)/'
  set_value: "<state> name <name>"
  default_value: ~

shutdown:
  get_value: '/^shutdown$/'
  set_value: "<state> shutdown"
  default: false
```